ddtrace/span.py

```diff
@@ -97,8 +94,12 @@ def __init__(
         self.duration_ns = None
 
         # tracing
-        self.trace_id = trace_id or _new_id()
-        self.span_id = span_id or _new_id()
+        # DEV: explicitly invoke ``__next__`` on generator vs using next(rng) as there is an overhead
+        # (my guess is that the overhead is for exception handling when using next(rng))
+
+        rng = rand.get_rand64_gen()
+        self.trace_id = trace_id or rng.__next__()
```

Comment:
If you want to optimize this further:

```python
if not trace_id and not span_id:
    trace_id = span_id = rng.__next__()
elif not trace_id:
    trace_id = rng.__next__()
elif not span_id:
    span_id = rng.__next__()
```

e.g. share the same id for the trace id and span id if it is a new root span. One less id to generate.

Reply:
we might be able to save this optimization for later, be curious to track benchmarks for "span start", both with no ids given, just trace id, just span id, and both trace and span id's given.